app/components/govuk_component/task_list_component/item_component.rb

```diff
@@ -0,0 +1,45 @@
+module GovukComponent
+  class TaskListComponent::ItemComponent < GovukComponent::Base
+    renders_one :title, "GovukComponent::TaskListComponent::TitleComponent"
+    renders_one :status, "GovukComponent::TaskListComponent::StatusComponent"
+
+    attr_reader :title_text, :status, :href, :hint
+
+    def initialize(title: nil, href: nil, hint: nil, status: nil, classes: [], html_attributes: {})
+      @title_text  = title
+      @href        = href
+      @hint        = hint
+      @status      = status
+
+      super(classes: classes, html_attributes: html_attributes)
+    end
+
+    def call
+      tag.li(safe_join([title_content, hint_content, status_content]), **html_attributes)
+    end
+
+  private
+
+    def title_content
+      title || with_title(text: title_text, href: href, hint: hint)
+    end
+
+    def status_content
+      case status
+      when String
+        with_status(text: status)
+      when Hash
+        with_status(**status)
+      else status
+      end
+    end
+
+    def hint_content
+      tag.div(hint, class: %w(govuk-task-list__task_hint))
```

Comment:
The hint is a bit more complicated, as it needs an `id` attribute setting, and then that ID should get added to the `aria-describedby` attribute on the link (space separated before the ID for the status div).

See [nunjucks code](https://github.com/alphagov/govuk-frontend/blob/main/packages/govuk-frontend/src/govuk/components/task-list/template.njk#L11) for reference.